### ISSUE TITLE
AssertionError when hitting ^C at REPL

### ISSUE THREAD

**Issue by ScottPJones** [JuliaLang/julia#13664]
Hitting ^C when a large array was being printed got this and exited Julia:
>9139685 -0.5601673648978617 -0.5526027080383488 -0.5450193446377157 -0.5374175314056321 -0.529797525676474 -0.5221595854003906 -0.5145039691345336 -0.5068309360346411 -0.49914074584585366 -0.4914336588942585 -0.4837099360777766 -0.4759698388575941 -0.468213629249237 -0.4604415698134733 -0.4526539236477281 -0.44485095437714034 -0.43703292614540923 -0.4292001036061587 -0.4213527519136716^C ERROR: AssertionError: d != C_NULL
 in uv_writecb_task at ./stream.jl:1004
 in process_events at ./stream.jl:713
 in wait at ./task.jl:372
 in wait at ./task.jl:286
 in wait at ./channels.jl:93
 in take! at ./channels.jl:82
 in send_to_backend at ./REPL.jl:611
 in send_to_backend at ./REPL.jl:608
 in anonymous at ./REPL.jl:622

**Comment by ScottPJones** [comment]
```
Julia Version 0.4.0
Commit 0ff703b* (2015-10-08 06:20 UTC)
Platform Info:
  System: Linux (x86_64-linux-gnu)
  CPU: Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Haswell)
  LAPACK: libopenblas64_
  LIBM: libopenlibm
  LLVM: libLLVM-3.7.0
```

**Comment by stevengj** [comment]
Seems like a dup of #2622.

**Comment by ScottPJones** [comment]
Thanks - however, this seems like it could happen with any I/O operation,not just printing something at the REPL.  How then does one develop applications in Julia that can't just arbitrarily exit if there is a ^C interrupt?

**Comment by stevengj** [comment]
One can't.  Ctrl-C isn't remotely safe until #2622 is resolved, because it's virtually impossible to predict when you might be in interrupt-unsafe C code.

**Comment by ScottPJones** [comment]
Ugh!  So absolutely no user facing code can be written in Julia currently, it seems like (if it needs to be reliable and not possible die from just a ^C).  Is the only way currently to just have the front end be something like a JS app?

**Comment by stevengj** [comment]
Well, you can just disable ctrl-c interrupts entirely.

**Comment by vtjnash** [comment]
fixed by #16174
